### ISSUE TITLE
Admin\Operator - External user can add an operator

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-registration#1960]
#### Dependencies

- Blocked by #1367 , https://github.com/bcgov/cas-registration/issues/2047
- Blocking

#### Description:

The operator form has changed since reg part 1, so we can't just copy/paste the reg 1 files over.

#### Acceptance Criteria:

Given I am an external user
When I add a new operator during the select operator workflow
Then I see the latest operator form
[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18028-10223&t=lpOQygv6eri4b1DZ-1)

Given I am an external user
When I click submit
Then I see the confirmation message/page

#### Development Checklist:

**API: Create Ninja Schema**
- [x] In `cas-registration/bc_obps/registration/schema/v2/operator.py` create `class OperatorIn(ModelSchema)`
- [x] In `cas-registration/bc_obps/registration/schema/v2/operator.py` create `class OperatorOutModelSchema)`
- [x] In `cas-registration/bc_obps/registration/schema/v2/user_operator.py` create `class UserOperatorOutModelSchema)`

**API: Create Service**
- [x] In `bc_obps/service/data_access_service/operator_service.py` create data access service to handle POST operator
- [x] In `cas-registration/bc_obps/service/operator_service_v2.py` create input and output service to handle POST operator `class OperatorServiceV2\ def create_operator...`

**API: Create Endpoint**
- [x] In `bc_obps/registration/api/v2/_user_operators/operators.py` similar to `bc_obps/registration/api/v1/user_operators.py`

**API: Add Tests**
- [x] In`bc_obps/registration/tests/endpoints/v2/_user_operators/test_current_v2.py` add tests for endpoint `@router.post(
    "/v2/user-operators",...`
- [x] In `bc_obps/service/operator_service_v2.py` add tests for services ` def create_operator...` etc.

**FE**
- [x] Create add operator route `bciers/apps/administration/app/bceidbusiness/industry_user/select-operator/(request-access)/add-user-operator/page.tsx` similar to `bciers/apps/registration1/app/(authenticated)/bceidbusiness/industry_user/dashboard/select-operator/(request-access)/user-operator/[id]/[formSection]/page.tsx`
- [x]  In `bciers/apps/administration/app/components/operators/OperatorForm.tsx` ensure the ` const endpoint` POST url is correct
- [x] In `bciers/apps/administration/app/components/operators/OperatorForm.tsx` ensure the `const method` is correctly using `
  const [isCreatingState, setIsCreatingState] = useState(isCreating);`,, similar to `bciers/apps/administration/app/components/contacts/ContactForm.tsx`
- [x] In `bciers/apps/administration/tests/components/operators` add vitests as required
- [ ]  Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by shon-button** [comment]
Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/select-operator
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/select-operator/add-operator

**Comment by patriciarussellCAS** [comment]
Thanks @shon-button I assume I need to recreate this using a different user than bc-cas-dev? 
This is what I get on the add operator page: 

![Image](https://github.com/user-attachments/assets/b0bd1888-815a-45c2-8cb2-8fdb9c0f50d5)

**Comment by shon-button** [comment]
> Thanks @shon-button I assume I need to recreate this using a different user than bc-cas-dev? This is what I get on the add operator page:
> 
> ![Image](https://private-user-images.githubusercontent.com/94197323/372549695-b0bd1888-815a-45c2-8cb2-8fdb9c0f50d5.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4MDcxNzAsIm5iZiI6MTcyNzgwNjg3MCwicGF0aCI6Ii85NDE5NzMyMy8zNzI1NDk2OTUtYjBiZDE4ODgtODE1YS00NWMyLThjYjItOGZkYjljMGY1MGQ1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDE4MjExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyODg3MjdkMzBhNTgzNjdlMDFiN2JlMzIxMTE3ZDA4Zjk0MTA0Y2U0MDA0ZWI3ZjdmOGQwOGI0NTdmNjQwMmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HN6C3yc_Z9q--mVZKaTixY00_xYCZJ9e3cdKWQa5NMQ)

@patriciarussellCAS Apologies for the lack of specific reproduction steps, I have updated the "Reachable at" comment to reflect the id to use: `As bc-cas-dev-secondary,`

**Comment by patriciarussellCAS** [comment]
Review notes: @shon-button & @zoeyli-46 

1. Using bc-cas-dev-secondary, when I click on Add Operator, I get this form (see screenshot), which does not include partner information like it shows on the figma wireframe. Is this correct or am I still seeing an old form?
2. After filling out the form and clicking "Submit" I see a snackbar "all changes were successful", not a confirmation page. Is this correct? 
3. Should the user be taken back to the admin dashboard after they click submit? Or just stay on this page until they click "cancel" or use breadcrumbs?

![add operator](https://github.com/user-attachments/assets/6273ed18-5f22-49ed-bae6-ff9b4badf31b)

**Comment by zoeyli-46** [comment]
@patriciarussellCAS for point 1, did you select business structure "general partnership"? because the partnership fields only show up if the business structure is general partnership.

After they click submit, it should have the snackbar instead of a confirmation page, but it should take them to the BCIERS dashboard where the "select operator" is now replaced with "operator - action required" This is in the [prototype](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18028-10207&node-type=frame&t=VUBhJcNyFg6GARQc-0&scaling=scale-down&content-scaling=fixed&page-id=7692%3A2&starting-point-node-id=18028%3A10207&show-proto-sidebar=1), and to be implemented in ticket #2048 which is coming next sprint! https://github.com/orgs/bcgov/projects/123/views/16?filterQuery=tiles&pane=issue&itemId=80445747

**Comment by patriciarussellCAS** [comment]
Thanks @zoeyli-46 That makes sense re: general partnership. 

So the only thing that didn't work for me is that I was NOT sent back to the BCIERS dashboard. @shon-button is that something that's being addressed in a separate ticket or should that be part of this work?

**Comment by zoeyli-46** [comment]
I have it as an AC in ticket #2048 https://github.com/orgs/bcgov/projects/123/views/16?filterQuery=tiles&pane=issue&itemId=80445747
